centreon-certified/prometheus/prometheus-pushgateway-metrics-apiv2.lua

```diff
@@ -0,0 +1,613 @@
+function EventQueue:format_metric_host(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_host]: starting format event host.")
+  local event = self.sc_event.event
+  local sdesc = "host"
+
+  event.formated_event = {
+    prom_hname      = event.cache.host.name,
+    prom_sdesc      = sdesc,
+    prom_sdesc_url  = base64.encode(sdesc)
+  }
+  self.sc_logger:debug("[EventQueue:format_metric_host]: call format_metric ")
+  self:format_metric_event(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_host]: format metric host is finished ")
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+--------------------------------------------------------------------------------
+function EventQueue:format_metric_service(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_service]: starting format event service.")
+  local event = self.sc_event.event
+  local sdesc = event.cache.service.description
+
+  event.formated_event = {
+    prom_hname      = event.cache.host.name,
+    prom_sdesc      = sdesc,
+    prom_sdesc_url  = base64.encode(sdesc)
+  }
+  self.sc_logger:debug("[EventQueue:format_metric_service]: call format_metric ")
+  self:format_metric_event(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_service]: format metric service is finished ")
+end
+
+--------------------------------------------------------------------------------
+-- add_unit_info: add unit metadata to match openmetrics standard
+-- @param {string} label, the name of the metric
+-- @param {string} unit, the unit name
+-- @param {string} name, the name of the metric
+-- @return {string} data, the unit metadata information
+--------------------------------------------------------------------------------
+function EventQueue:add_unit_info (label, unit, name)
+  local data = ''
+
+  if (unit ~= '' and unit ~= nil) then
+      data = '# UNIT ' .. name .. '\n'
+  end
+
+  return data
+end
+
+--------------------------------------------------------------------------------
+-- create_metric_name: concatenates data to create the metric name
+-- @param {string} label, the name of the perfdata
+-- @param {string} unit, the unit name
+-- @return {string} name, the prometheus metric name (open metric format)
+--------------------------------------------------------------------------------
+function EventQueue:create_metric_name (label, unit)
+  local name = ''
+  local sdesc = 'host'
+  if (self.sc_event.event.service_description) then
+    sdesc = self.sc_event.event.service_description
+  end
+  local hname = self.sc_event.event.cache.host.name
+
+  if (unit ~= '') then
+    if (self.enable_extended_metric_name == 0) then
+      name = label .. '_' .. unit
+    else
+      name = hname .. '_' .. sdesc .. ':' .. label .. '_' .. unit
+    end
+  else
+    if (self.enable_extended_metric_name == 0) then
```

Comment:
this parameter doesn't exist. Must be added in the EventQueue constructor

Reply:
fixed there https://github.com/centreon/centreon-stream-connector-scripts/pull/279/commits/6e5bfd672680c331dbd2bd5cc87dfc84c2debd31